### ISSUE TITLE
Still forwarded to upstream (due to a valid cookie) even if the session was terminated in keycloak. (oauth2/auth endpoint)

### ISSUE THREAD

**Issue by Hadsch** [oauth2-proxy/oauth2-proxy#1800]
# Scenario
* Call `https://myapp.example` and log in with your credentials
* Logout from the session directly in keycloak
* Call `https://myapp.example` again

## Expected Behavior

Since the session is terminated already in the identity provider, I would expect the `oauth2/auth` endpoint is returning 401 and forces the user to log in again.

## Current Behavior

It seems, that because of the valid cookie the `oauth2/auth` provider is not checking if the sessions is valid and returns an 202 response.

### nginx
Use the nginx setup described here: https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview#configuring-for-use-with-the-nginx-auth_request-directive

Since I have SPA the only change I made is this: 
```
        try_files $uri $uri/ /index.html;
        # or "root /path/to/site;" or "fastcgi_pass ..." etc
```

### oauth2-proxy
```
    # Provider config
    provider="keycloak-oidc"
    redirect_url="https://dashboard.myapp.example.com/oauth2/callback"
    oidc_issuer_url="https://keycloak.myapp.example.com/auth/realms/che"
    ssl_insecure_skip_verify=true
    # Client config
    client_id="your-clientid"
    client_secret="your-secret"
    cookie_secret="your-cookie-secret"
    cookie_secure="false"
    # Upstream config: nginx uses oauth2-proxy for the 'auth_request' directive. If authenticated nginx will serve the app
    http_address="0.0.0.0:4180"
    upstreams="file:///dev/null"
    # Proxy Config
    skip_auth_routes=["/health.*"]
    skip_provider_button="true"
    reverse_proxy="true"
    email_domains=["*"]
    cookie_domains=["myapp.example.com"]
    whitelist_domains=[".myapp.example.com:*"]
```

# TLDR

After terminating a valid session in the identity-provider (e.g. keycloak) `/oauth2/auth` endpoint should return 401 even if a valid cookie is sent along with the request.

**Comment by Hadsch** [comment]
Is this normal behavior since the cookie is valid or is there something fishy with my config?

**Comment by t-katsumura** [comment]
@Hadsch 
Yes. This is the current behaviour of oauth2-proxy and there is no way to avoid this.
We faced the same problem and that's why I sent this PR https://github.com/oauth2-proxy/oauth2-proxy/pull/1751

**Comment by Hadsch** [comment]
Your PR seems to cover my needs, I hope I'll find some time in the next few days to try it out. 
Since I'm not really an oauth-expert it seems a bit awkward to me, that this isn't the default behavior!
